ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -753,6 +759,16 @@ func (r *RayJobReconciler) deleteClusterResources(ctx context.Context, rayJobIns
 		if !cluster.DeletionTimestamp.IsZero() {
 			logger.Info("The deletion of the associated RayCluster for RayJob is ongoing.", "RayCluster", cluster.Name)
 		} else {
+			if r.useBackgroundGoroutine {
+				rayDashboardClient, err := r.dashboardClientFunc(&cluster, rayJobInstance.Status.DashboardURL)
+				if err != nil {
+					logger.Error(err, "Failed to get dashboard client for RayJob")
+				}
+				// clear cache, and it will remove this job from the cache updating loop.
+				if err := rayDashboardClient.StopJob(ctx, rayJobInstance.Status.JobId); err != nil {
+					logger.Error(err, "Failed to stop job for RayJob")
+				}
```

Comment:
Is it okay to call StopJob to remove the cache placeholder before deleting the RayCluster?

Reply:
We probably should not do this. Just let the cache client figure out how to deal with old entries by itself.

Reply:
Fixed at https://github.com/ray-project/kuberay/pull/4160/commits/9f87da63f2ecb952e8d69a3b5327652dff4c8832. The the update of JobInfo would be remove from the updating loop if it reach the terminal status. Eventually, the cached JobInfo will be remove from cache as it elapses the expiration time. If it is not acceptable, kindly let me know.